src/toil/batchSystems/kubernetes.py

```diff
@@ -677,44 +679,55 @@ def getUpdatedBatchJob(self, maxWait):
-                                logger.info("REASON: %s Exit Code: %s", termination.reason, termination.exit_code)
-                                
-                                if termination.exit_code != 0:
-                                    # The pod failed. Dump information about it.
-                                    logger.debug('Failed pod information: %s', str(pod))
-                                    logger.warning('Log from failed pod: %s', self._getLogForPod(pod))
-                                jobID = int(pod.metadata.owner_references[0].name[len(self.jobPrefix):])
-                                terminated = pod.status.container_statuses[0].state.terminated
-                                runtime = slow_down((terminated.finished_at - terminated.started_at).total_seconds())
-                                result = UpdatedBatchJobInfo(jobID=jobID, exitStatus=terminated.exit_code, wallTime=runtime, exitReason=None)
-                                self._try_kubernetes(self._api('batch').delete_namespaced_job, 
-                                            pod.metadata.owner_references[0].name,
-                                            self.namespace,
-                                            propagation_policy='Foreground')
-
-                                self._waitForJobDeath(pod.metadata.owner_references[0].name)
-                                return result
-                            else:
-                                continue
+            for event in self._try_kubernetes_stream(self._api('batch').list_namespaced_job, self.namespace, 
+                                                        label_selector="toil_run={}".format(self.runID),
+                                                        timeout_seconds=maxWait):
+                # Grab the metadata data, ID and the list of conditions of the current job
+                jobObject = event['object']
+                jobID = int(jobObject.metadata.name[len(self.jobPrefix):])
+                jobObjectListConditions =jobObject.status.conditions
+                # Exit Reason defaults to 'Successfully Finsihed` unless said otherwise 
+                ExitReason = 1
+                ExitCode = 0
+
+                # Check if jobs exists by seeing if condition logs are pressent
+                if jobObjectListConditions is None or len(jobObjectListConditions):
+                    logger.debug("No job container statuses for job %s" % (jobObject.metadata.name))
+                    return UpdatedBatchJobInfo(jobID=jobID, exitStatus=EXIT_STATUS_UNAVAILABLE_VALUE, wallTime=0, exitReason=3)
+                
+                # Check if there are any active pods
+                if jobObject.status.acitve > 0:
+                    logger.info("%s has %d pods running" % jobObject.metadata.name, jobObject.status.active)
+                    continue
+                else:
+                    # No more active pods in the current job ; must be finished
```

Comment:
Is this true? What happens if the pod is pending because it can't run right now because the cluster is full? Does the job still show it as "active"? What if the job has been created but Kubernetes hasn't yet created a pod for it?

I think you should use `jobObject.status.finished` and `jobObject.status.failed` instead. if either of those is 1, a pod has definitely finished or failed for the job. But there are a variety of reasons a pod might not be "actively running", even though it hasn't finished or failed yet.